fs/fs.go

```diff
@@ -439,12 +442,19 @@ func (fs *filesystem) Unmount(ctx context.Context, mountpoint string) error {
 	l.Done()
 	fs.layerMu.Unlock()
 	fs.metricsController.Remove(mountpoint)
-	// The goroutine which serving the mountpoint possibly becomes not responding.
-	// In case of such situations, we use MNT_FORCE here and abort the connection.
-	// In the future, we might be able to consider to kill that specific hanging
-	// goroutine using channel, etc.
-	// See also: https://www.kernel.org/doc/html/latest/filesystems/fuse.html#aborting-a-filesystem-connection
-	return syscall.Unmount(mountpoint, syscall.MNT_FORCE)
+
+	if err := unix.Unmount(mountpoint, 0); err != nil {
+		if err != unix.EBUSY {
+			return err
```

Comment:
Do we need to retry on EINTR?

Reply:
Thanks for the review. Added retrying on EINTR.